packages/yoga/src/Tag/native/Tag.jsx

```diff
@@ -1,115 +1,74 @@
 import React from 'react';
 import styled, { withTheme } from 'styled-components';
-import { func, oneOf, oneOfType, bool, node } from 'prop-types';
-import { hexToRgb } from '@gympass/yoga-common';
-import { TriangleAlert } from '@gympass/yoga-icons';
+import { oneOf, node } from 'prop-types';
+import Text from '../../Text';
 
-import { Text } from '../..';
+export const StyledTag = styled.View`
+  justify-content: center;
+  align-items: center;
 
-const StyledTag = styled.View`
   ${({
-    color,
-    full,
+    variant,
     theme: {
       yoga: {
+        colors: {
+          elements,
+          feedback: {
+            [variant]: borderColor = { dark: elements.selectionAndIcons },
+          },
+        },
         components: { tag },
       },
     },
   }) => `
-    width: ${full ? '100%' : 'auto'};
+    width: auto;
     padding:
       ${tag.padding.top}px
       ${tag.padding.right}px
       ${tag.padding.bottom}px
       ${tag.padding.left}px;
 
-    background-color: ${hexToRgb(color, 0.25)};
     border-radius: ${tag.border.radius}px;
+    border-width: ${tag.border.width}px;
+    border-color: ${borderColor.dark};
   `}
 `;
 
-const Wrapper = styled.View`
-  flex-direction: row;
-  justify-content: center;
-  align-items: center;
-`;
-
-const StyledText = styled(Text.Bold)`
+export const StyledText = styled(({ variant, ...rest }) => <Text {...rest} />)`
   ${({
-    color,
+    variant,
     theme: {
       yoga: {
+        colors: {
+          text,
+          feedback: { [variant]: color = { dark: text.secondary } },
+        },
         components: { tag },
       },
     },
   }) => `
-    color: ${color};
+    color: ${color.dark};
 
     font-size: ${tag.font.size}px;
+    line-height: ${tag.font.lineHeight}px;
+    font-weight: ${tag.font.weight};
   `}
 `;
 
 /** Use Tag component when you want to categorize your content */
-const Tag = ({
-  icon: Icon,
-  children,
-  variant,
-  theme: {
-    yoga: {
-      components: { tag },
-      colors: { text, [variant]: color = text.secondary },
-    },
-  },
-  ...props
-}) => (
-  <StyledTag color={color} {...props}>
-    <Wrapper>
-      {Icon && (
-        <Icon
-          width={14}
-          height={12}
-          fill={color}
-          style={{ marginRight: tag.icon.margin.right + 1 }}
-        />
-      )}
-
-      <StyledText color={color}>{children}</StyledText>
-    </Wrapper>
+const Tag = ({ children, ...props }) => (
+  <StyledTag {...props}>
+    <StyledText {...props}>{children}</StyledText>
```

Comment:
```suggestion
const Tag = ({ children, variant, ...props }) => (
  <StyledTag variant={variant} {...props}>
    <StyledText variant={variant}>{children}</StyledText>
```

We will face some weird errors if you keep these two `{...props}` in `StyledTag` and `StyledText`.

Per exemple:
If someone wants to add a margin in this component, like this:
```jsx
<Tag style={{ marginRight: 28 }} />
```
It will be added on the `StyledTag` and in `StyledText`

Reply:
Done!